framework/auth/views.py

```diff
@@ -202,87 +202,157 @@ def forgot_password_post(auth, **kwargs):
     return {}
 
 
-@collect_auth
-def auth_login(auth, **kwargs):
+def login_and_register_handler(auth, login=True, campaign=None, next_url=None, logout=None):
     """
-    This view serves as the entry point for OSF login and campaign login.
-    HTTP Method: GET
+    Non-view helper to handle `login` and `register` requests.
 
-        GET '/login/' without any query parameter:
-            redirect to CAS login page with dashboard as target service
+    :param auth: the auth context
+    :param login: `True` if `GET /login`, `False` if `GET /register`
+    :param campaign: a target campaign defined in `auth.campaigns`
+    :param next_url: the service url for CAS login or redirect url for OSF
+    :param logout: used only for `claim_user_registered`
+    :return: data object that contains actions for `auth_register` and `auth_login`
+    :raises: http.BAD_REQUEST
+    """
 
-        GET '/login/?logout=true
-            log user out and redirect to CAS login page with redirect_url or next_url as target service
+    # set target page to `dashboard` if no service url is specified
+    if not next_url:
```

Comment:
`next_url` is set to default `dashboard` when not provided, which guarantees `not None`